src/olympia/search/filters.py

```diff
@@ -105,6 +112,72 @@ def filter_queryset(self, request, qs, view):
             should=primary_should + secondary_should), functions=functions)
 
 
+class SearchParameterFilter(BaseFilterBackend):
+    """
+    A django-rest-framework filter backend that filters only items in an ES
+    query that match a specific set of fields: app, platform and type.
+    """
+    # FIXME: add appversion.
+
+    Filter = namedtuple(
+        'Filter', ['query_param', 'valid_values', 'reverse', 'es_field'])
```

Comment:
hmm, could we define `Filter` outside  of this class to allow easier reuse? Also, the `field_names` argument can easily be a tuple (nitpick).